backend/domain_types/src/types.rs

```
@@ -221,6 +222,32 @@ impl<
                 grpc_api_types::payments::payment_method::PaymentMethod::Reward(_reward) => {
                     Ok(PaymentMethodData::Reward)
                 }
+                grpc_api_types::payments::payment_method::PaymentMethod::Wallet(wallet_type) => {
+                    match wallet_type.wallet_type {
+                        Some(grpc_api_types::payments::wallet_payment_method_type::WalletType::Mifinity(mifinity_data)) => {
+                            Ok(PaymentMethodData::Wallet(crate::payment_method_data::WalletData::Mifinity(
+                                crate::payment_method_data::MifinityData {
+                                    date_of_birth: hyperswitch_masking::Secret::new(
+                                        time::Date::parse(&mifinity_data.date_of_birth, &time::format_description::well_known::Iso8601::DATE)
+                                            .map_err(|_| ApplicationErrorResponse::BadRequest(ApiError {
```

don't discard the error, log the error